### PR TITLE
Better AddonLifecycle AddonEvent docs

### PR DESCRIPTION
**Pull request by MidoriKami**
@KazWolfe Yo, I heard you like docs, so I added some docs to your docs.

### CODE REVIEW

File: Dalamud/Game/Addon/Lifecycle/AddonEvent.cs [hunk 1]
```diff
@@ -6,67 +6,86 @@
 public enum AddonEvent
 {
     /// <summary>
-    /// Event that is fired before an addon begins it's setup process.
+    /// Event that is fired before an addon begins its setup process.
     /// </summary>
+    /// <remarks>Setup occurs when an addon is constructing its ui node lists, and registering event callbacks.</remarks>
```
Comment (KazWolfe) [inline Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:11]: Please try, where possible, to keep to a single `<remarks>` element. You can use `<br />` for newlines.

**Review by KazWolfe — DISMISSED** [review]

File: Dalamud/Game/Addon/Lifecycle/AddonEvent.cs [hunk 2]
```diff
@@ -1,72 +1,121 @@
...
+    /// <br />
+    /// As this is part of the destruction process for an addon, this event does not have an associated Post event.
+    /// </remarks>
+    /// <seealso cref="AddonFinalizeArgs"/>
     PreFinalize,
     
     /// <summary>
-    /// Event that is fired before an addon begins a requested update.
+    /// An event that is fired before an addon begins a requested update via
+    /// <see cref="AtkUnitBase.OnRequestedUpdate"/>. Requested updates generally occur when the server sends new data to
```
Comment (Haselnussbomber) [inline Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:70]: I know you use this explanation a lot, but you’re completely leaving out the fact that this is triggered by subscribed AtkArrayData.

File: Dalamud/Game/Addon/Lifecycle/AddonEvent.cs [hunk 3]
```diff
@@ -1,72 +1,121 @@
...
-    /// Event that is fired after an addon finishes a requested update.
+    /// An event that is fired after an addon has finished processing a requested update.
+    /// See <see cref="PreRequestedUpdate"/> for more information.
     /// </summary>
     PostRequestedUpdate,
     
     /// <summary>
-    /// Event that is fired before an addon begins a refresh.
+    /// An event that is fired before an addon begins processing a refresh via
+    /// <see cref="AtkUnitManager.RefreshAddon"/>. Refreshes are generally triggered in response to certain user
```
Comment (Haselnussbomber) [inline Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:92]: Here you could add that it’s used to update the addons AtkValues.

File: Dalamud/Game/Addon/Lifecycle/AddonEvent.cs (hunk 2 above)
Comment (MidoriKami) [reply Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:70]: How do you suggest wording this?

Comment (Haselnussbomber) [reply Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:70]: I don't know, keeping it simple?
```cs
/// <summary>
/// An event fired before an addon begins a requested update via <see cref="AtkUnitBase.OnRequestedUpdate"/>.
/// Requested updates occur when data in the subscribed AtkArrayData has changed.
/// </summary>
```

File: Dalamud/Game/Addon/Lifecycle/AddonEvent.cs [hunk 4]
```diff
@@ -1,72 +1,121 @@
...
+    /// developers seeking to add custom elements to now-initialized and populated node lists, as well as reading data
+    /// placed in the AtkValues by the game during the setup process.
+    /// See <see cref="PreSetup"/> for more information.
     /// </summary>
     PostSetup,
 
     /// <summary>
-    /// Event that is fired before an addon begins update.
+    /// An event that is fired before an addon begins its update cycle via <see cref="AtkUnitBase.Update"/>. This event
+    /// is fired every frame that an update is loaded, regardless of visibility.
```
Comment (Haselnussbomber) [inline Dalamud/Game/Addon/Lifecycle/AddonEvent.cs:30]: I think the wording is a bit weird here.
```suggestion
    /// An event that is fired before an addon begins its update cycle via <see cref="AtkUnitBase.Update"/>. This event
    /// is fired every frame, regardless of visibility.
```

**Comment by goaaats** [comment]
Is this good to go?

**Comment by MidoriKami** [comment]
I want to add https://github.com/goatcorp/Dalamud/pull/1964#discussion_r1690026629 but other than that it's good to go

**Review by KazWolfe — APPROVED** [review]

**Comment by Haselnussbomber** [comment]
> I want to add [#1964 (comment)](https://github.com/goatcorp/Dalamud/pull/1964#discussion_r1690026629) but other than that it's good to go
> 
> Won't be able to add that in for a couple hours.

Kaz already added this in [7fff487](https://github.com/goatcorp/Dalamud/pull/1964/commits/7fff48752e224b5ec3b1267de30d33c155373df1) :)

**Comment by MidoriKami** [comment]
Oh didn't see that, hard to read on mobile. Nice.

**Pull request merged**